app/forms/declaration/PackageInformation.scala

```diff
@@ -17,50 +17,84 @@
 
-  def require1Field[T](fs: (T => Option[_])*): T => Boolean =
-    t => fs.exists(f => f(t).nonEmpty)
+  def fromJsonString(value: String): Option[PackageInformation] = Json.fromJson(Json.parse(value)).asOpt
 
   implicit val format = Json.format[PackageInformation]
 
   val formId = "PackageInformation"
   val limit = 99
 
-  val mapping = Forms
+  private val NumberOfPackagesLimitLower = 0
+  private val NumberOfPackagesLimitUpper = 99999
+
+  val mappingAllFieldsMandatory = Forms
     .mapping(
       "typesOfPackages" ->
         text()
           .verifying("supplementary.packageInformation.typesOfPackages.empty", nonEmpty)
           .verifying("supplementary.packageInformation.typesOfPackages.error", isEmpty or isContainedIn(PackageTypes.all.map(_.code))),
       "numberOfPackages" ->
         number()
-          .verifying("supplementary.packageInformation.numberOfPackages.error", q => q > 0 && q <= 999999),
+          .verifying("supplementary.packageInformation.numberOfPackages.error", isInRange(NumberOfPackagesLimitLower, NumberOfPackagesLimitUpper)),
       "shippingMarks" ->
         text()
           .verifying("supplementary.packageInformation.shippingMarks.empty", nonEmpty)
           .verifying("supplementary.packageInformation.shippingMarks.characterError", isEmpty or isAlphanumericWithAllowedSpecialCharacters)
           .verifying("supplementary.packageInformation.shippingMarks.lengthError", isEmpty or noLongerThan(42))
+    )(PackageInformation.fromValues)(PackageInformation.toValues)
+
+  val mappingAllFieldsOptional = Forms
+    .mapping(
+      "typesOfPackages" -> optional(
+        text()
+          .verifying("supplementary.packageInformation.typesOfPackages.error", isContainedIn(PackageTypes.all.map(_.code)))
+      ),
+      "numberOfPackages" -> optional(
+        number()
+          .verifying("supplementary.packageInformation.numberOfPackages.error", isInRange(NumberOfPackagesLimitLower, NumberOfPackagesLimitUpper))
+      ),
+      "shippingMarks" -> optional(
+        text()
+          .verifying("supplementary.packageInformation.shippingMarks.characterError", isAlphanumericWithAllowedSpecialCharacters)
+          .verifying("supplementary.packageInformation.shippingMarks.lengthError", noLongerThan(42))
+      )
     )(PackageInformation.apply)(PackageInformation.unapply)
+    .verifying("supplementary.packageInformation.empty", validatePackageInformation(_))
 
-  val DUPLICATE_MSG_KEY = "supplementary.packageInformation.global.duplicate"
-  val LIMIT_MSG_KEY = "supplementary.packageInformation.global.limit"
-  val USE_ADD = "supplementary.packageInformation.global.useAdd"
-  val ADD_ONE = "supplementary.packageInformation.global.addOne"
+  private def validatePackageInformation(packageInformation: PackageInformation): Boolean =
+    !(packageInformation.typesOfPackages.isEmpty && packageInformation.numberOfPackages.isEmpty && packageInformation.shippingMarks.isEmpty)
```

Comment:
I think it can be useful to have method `isEmpty` inside the PackageInformation class and just invoke `_.isEmpty`